lib/hooks/LocalStorage/index.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+
+/**
+Hook to store a persistent state in local storage.
+
+The state is shared among any component using it
+from any tab of the same browser
+accessing any application hosted in the same domain.
+
+A null value means the value is unset.
+Setting the state to null unsets it.
+Any other value is converted to string including undefined.
+The initial value is set only if the value is unset
+on component initialization.
+Initial value of undefined is ignored.
+*/
+
+export default function useLocalStorage(key, initialValue) {
+  function decode(json) {
```

Comment:
Do you think include a try-catch block when using JSON.parse and JSON.stringify to avoid crashing in case of malformed data would be worth? 

Example:
```javascript
  const decode = (json) => {
    try {
      return json !== null ? JSON.parse(json) : null
    } catch (error) {
      console.error('Error parsing JSON from localStorage:', error)
      return null
    }
  }
```